src/prng/chacha.rs

```diff
@@ -154,22 +148,15 @@ impl ChaChaRng {
     /// ```
     /// [1]: Daniel J. Bernstein. [*Extending the Salsa20
     /// nonce.*](https://cr.yp.to/papers.html#xsalsa)
-    fn init(&mut self, key: &[u32; KEY_WORDS]) {
-        self.state[0] = 0x61707865;
-        self.state[1] = 0x3320646E;
-        self.state[2] = 0x79622D32;
-        self.state[3] = 0x6B206574;
-
-        for i in 0..KEY_WORDS {
-            self.state[4+i] = key[i];
-        }
-
-        self.state[12] = 0;
-        self.state[13] = 0;
-        self.state[14] = 0;
-        self.state[15] = 0;
-
-        self.index = STATE_WORDS;
+    fn init(seed: [u32; SEED_WORDS]) -> Self {
+        ChaChaRng {
+            buffer: [0; STATE_WORDS],
+            state: [0x61707865, 0x3320646E, 0x79622D32, 0x6B206574, // constants
+                    seed[0], seed[1], seed[2], seed[3], // seed
+                    seed[4], seed[5], seed[6], seed[7], // seed
+                    0, 0, 0, 0], // counter
+            index: STATE_WORDS, // generate on first use
+         }
```

Comment:
Was missing these changes from https://github.com/dhardy/rand/pull/77, and they make `from_seed` easier.

Reply:
It does not do all that much. The old code created an empty struct first, and than called `init` to fill it. Now `init` creates the struct itself.